src/clp_ffi_py/ir/native/PyDeserializer.cpp

```diff
@@ -0,0 +1,281 @@
+#include <clp_ffi_py/Python.hpp>  // Must always be included before any other header files
+
+#include "PyDeserializer.hpp"
+
+#include <system_error>
+#include <type_traits>
+#include <utility>
+
+#include <clp/ErrorCode.hpp>
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/ir_stream/Deserializer.hpp>
+#include <clp/ffi/ir_stream/IrUnitType.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/time_types.hpp>
+#include <clp/TraceableException.hpp>
+
+#include <clp_ffi_py/api_decoration.hpp>
+#include <clp_ffi_py/error_messages.hpp>
+#include <clp_ffi_py/ExceptionFFI.hpp>
+#include <clp_ffi_py/ir/native/DeserializerBufferReader.hpp>
+#include <clp_ffi_py/ir/native/error_messages.hpp>
+#include <clp_ffi_py/ir/native/PyKeyValuePairLogEvent.hpp>
+#include <clp_ffi_py/PyObjectCast.hpp>
+#include <clp_ffi_py/utils.hpp>
+
+namespace clp_ffi_py::ir::native {
+using clp::ffi::ir_stream::IRErrorCode;
+using clp::ffi::ir_stream::IrUnitType;
+
+namespace {
+/**
+ * Callback of `PyDeserializer`'s `__init__` method:
+ */
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+PyDoc_STRVAR(
+        cPyDeserializerDoc,
+        "Deserializer for deserializing CLP key-value pair IR streams.\n"
+        "This class deserializes log events from a CLP key-value pair IR stream.\n\n"
+        "__init__(self, input_stream, buffer_capacity=65536, allow_incomplete_stream=False)\n\n"
+        "Initializes a :class:`Deserializer` instance with the given inputs. Note that each"
+        " object should only be initialized once. Double initialization will result in a memory"
+        " leak.\n\n"
+        ":param input_stream: Input stream that contains serialized CLP IR.\n"
```

Comment:
Perhaps you don't need to double mentioning "input stream" since the variable name already explains itself?
```suggestion
        ":param input_stream: serialized CLP IR stream.\n"
```